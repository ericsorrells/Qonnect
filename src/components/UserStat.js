// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
// ========================================================================================

const UserStat = ({ stat, title }) => {
  return (
    <div className="user-stat">
      <div className="user-stat__stat" >{stat}</div>
      <div className="user-stat__title">{title}</div>
    </div>
  )
}

UserStat.propTypes = {
  stat:  PropTypes.number,
  title: PropTypes.string
}

export default UserStat;