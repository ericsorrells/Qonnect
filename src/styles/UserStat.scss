.user-stat {
  margin-top: 20px;
  margin-bottom: 20px;
}

.user-stat__stat {
  font-size: $user-data-font;
  color: $main-color;
  font-weight: bold;
}

.user-stat__title {
  font-size: .75rem;
  color: $dark-grey;
}