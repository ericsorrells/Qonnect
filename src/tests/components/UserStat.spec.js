// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockStoreSetup from '../../utils/mockStoreHelpers';
import 'jest-enzyme';
// ========================================================================================
import UserStat from '../../components/UserStat';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<UserStat/>', () => {
  it('renders correctly', () => {
    const component = shallow(
      <UserStat
        stat={43}
        title={'Some Stat'}
      />
    )

    expect(component).toMatchSnapshot();
  });
});
