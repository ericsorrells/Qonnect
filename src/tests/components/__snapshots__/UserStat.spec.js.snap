// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStat/> renders correctly 1`] = `
<div
  className="user-stat"
>
  <div
    className="user-stat__stat"
  >
    43
  </div>
  <div
    className="user-stat__title"
  >
    Some Stat
  </div>
</div>
`;
